java/kafka/hello-world-consumer/src/main/java/KafkaConsumerExample.java

```diff
@@ -44,6 +45,12 @@ public static void main(String[] args) {
                 log.info("\tpartition: {}", record.partition());
                 log.info("\toffset: {}", record.offset());
                 log.info("\tvalue: {}", record.value());
+                if (record.headers() != null) {
+                    log.info("\theaders: ");
+                    for (Header header : record.headers().toArray()) {
```

Comment:
is this `toArray()` really needed?

Reply:
Not at all - as you can see, the log is not correctly parsed:
```
2020-07-23 16:06:59 INFO  KafkaConsumerExample:44 - Received message:
2020-07-23 16:06:59 INFO  KafkaConsumerExample:45 - 	partition: 0
2020-07-23 16:06:59 INFO  KafkaConsumerExample:46 - 	offset: 40
2020-07-23 16:06:59 INFO  KafkaConsumerExample:47 - 	value: "Hello world - 40"
2020-07-23 16:06:59 INFO  KafkaConsumerExample:49 - 	headers: 
2020-07-23 16:06:59 INFO  KafkaConsumerExample:51 - 		key: foo, value: bar, sba
```
The headers are `foo=bar, sba=asd`, so I think it's correct that I added `toArray()`.

Reply:
And it's not even on the next line:
```
2020-07-23 16:06:57 INFO  KafkaConsumerExample:44 - Received message:
2020-07-23 16:06:57 INFO  KafkaConsumerExample:45 - 	partition: 0
2020-07-23 16:06:57 INFO  KafkaConsumerExample:46 - 	offset: 38
2020-07-23 16:06:57 INFO  KafkaConsumerExample:47 - 	value: "Hello world - 38"
2020-07-23 16:06:57 INFO  KafkaConsumerExample:49 - 	headers: 
2020-07-23 16:06:57 INFO  KafkaConsumerExample:51 - 		key: foo, value: bar, sba
2020-07-23 16:06:58 INFO  KafkaConsumerExample:44 - Received message:
2020-07-23 16:06:58 INFO  KafkaConsumerExample:45 - 	partition: 0
2020-07-23 16:06:58 INFO  KafkaConsumerExample:46 - 	offset: 39
2020-07-23 16:06:58 INFO  KafkaConsumerExample:47 - 	value: "Hello world - 39"
2020-07-23 16:06:58 INFO  KafkaConsumerExample:49 - 	headers: 
2020-07-23 16:06:58 INFO  KafkaConsumerExample:51 - 		key: foo, value: bar, sba
2020-07-23 16:06:59 INFO  KafkaConsumerExample:44 - Received message:
2020-07-23 16:06:59 INFO  KafkaConsumerExample:45 - 	partition: 0
2020-07-23 16:06:59 INFO  KafkaConsumerExample:46 - 	offset: 40
2020-07-23 16:06:59 INFO  KafkaConsumerExample:47 - 	value: "Hello world - 40"
2020-07-23 16:06:59 INFO  KafkaConsumerExample:49 - 	headers: 
2020-07-23 16:06:59 INFO  KafkaConsumerExample:51 - 		key: foo, value: bar, sba
```

Reply:
why formatting should be different if what you get is always an `Header` instance using or not `toArray()` ?

Reply:
I don't know, gonna try that one more time. Maybe there is problem somewhere else in my code :)

Reply:
I found mistake in my regex, because when I set the env to `foo=bar, sba=asd`, it took this whole string, as any `\n\r\t` was present. So that was my fault! Gonna fix it, build it, run it and let you know. Sorry @ppatierno

Reply:
Done